README.md

```diff
@@ -1,3 +1,215 @@
+linux compiler used. Other branches may be for the the llvm tools which are needed on all platforms. It is best to check
+all the different runtimecore branches.
+
+The linux clang compiler is also a custom build. This is due to the need of using libc++ and libc++abi instead of
+libstdc++. Since some of our products link to libstdc++, it is crucial to statically link in these runtimes and to hide
+all symbols in order to allow using both runtimes in one application.
+
+To get setup for the build, clone the right sources:
+
+```sh
+mkdir -p ${HOME}/llvm && cd ${HOME}/llvm
+git clone --branch runtimecore_19.1.2 git@github.com:Esri/include-what-you-use.git
+git clone --branch runtimecore_19.1.2 git@github.com:Esri/llvm-project.git
+```
+
+## Linux
+
+### Building the Linux Compiler
+
+To build on Linux, you MUST use the [llvm.dockerfile](llvm.dockerfile) dockerfile to build. This will give you the
+correct ubuntu 22.04 environment that is needed to build with a compatible ABI. This matches our lowest supported
+platform for RTC and also makes sure that we're building with a clean environment C runtime. Using docker will also more
+easily allow you to build for arm64 using an arm64 based mac machine to get good compilation times. To build the images,
+you can run the following commands:
+
+```sh
+docker build --platform=linux/amd64 --tag=llvm:19.1.2-amd64 - < ${HOME}/llvm/llvm-project/llvm.dockerfile
+docker build --platform=linux/arm64 --tag=llvm:19.1.2-arm64 - < ${HOME}/llvm/llvm-project/llvm.dockerfile
+```
+
+This will create the needed Ubuntu 22.04 sand-boxed environments that we'll use to build. When you're ready to build,
+you can then start the container with the following command:
+
+`docker run --rm -it -u $(id -u):$(id -g) --volume ${HOME}/llvm:/llvm --workdir /llvm llvm:19.1.2-amd64 bash`
+
+You'll now be in the container environment bash and can run the cmake commands to build. The following Cmake commands
+are very similar to the [Fushsia](clang/cmake/caches/Fuchsia.cmake) commands. The commands will build the llvm compiler
+as well as all the runtimes that will be statically linked into RTC so the library can be as portable as possible and
+can run on applications that use the `libstdc++` runtime.
+
+```sh
+# ${target} is set at the dockerfile for convenience
+cmake -S llvm-project/llvm -B /llvm/${target}/build -G "Ninja" \
+  -DCMAKE_AR="/usr/bin/llvm-ar" \
+  -DCMAKE_BUILD_TYPE="Release" \
+  -DCMAKE_C_COMPILER="/usr/bin/clang" \
+  -DCMAKE_C_COMPILER_LAUNCHER="/usr/bin/ccache" \
+  -DCMAKE_CXX_COMPILER="/usr/bin/clang++" \
+  -DCMAKE_CXX_COMPILER_LAUNCHER="/usr/bin/ccache" \
+  -DCMAKE_INSTALL_PREFIX="/llvm/${target}/19.1.2" \
+  -DCMAKE_RANLIB="/usr/bin/llvm-ranlib" \
+  \
+  -DLLVM_BUILTIN_TARGETS="${target}" \
+  -DLLVM_ENABLE_LTO="Thin" \
+  -DLLVM_ENABLE_PROJECTS="clang;clang-tools-extra;lld" \
+  -DLLVM_ENABLE_RUNTIMES="compiler-rt;libcxx;libcxxabi;libunwind" \
+  -DLLVM_EXTERNAL_IWYU_SOURCE_DIR="/llvm/include-what-you-use" \
```

Comment:
@cfrankmiller in thise case no, since we're building in a container and the ${HOME} variable is actually empty. Funny story though, I had originally added ${HOME} here which would then evaluate this path to `//llvm/` all over the place. The llvm CMake build system must have something that looks for paths using `/` separators because I was then hit with build errors in the builtin library. I will probably create a repro case for them to see if they can fix things as it set me back a week to figure out what went wrong.